src/main/java/ch/njol/skript/effects/EffPlaySound.java

```diff
@@ -47,186 +46,136 @@
-				if (soundEnum == null) {
-					sound = sound.toLowerCase(Locale.ENGLISH);
-					if (!SOUND_VALID_PATTERN.matcher(sound).matches())
-						continue;
-					w.playSound(location, sound, category, volume, pitch);
-				} else {
-					w.playSound(location, soundEnum, category, volume, pitch);
-				}
-			} else {
-				if (soundEnum == null) {
-					sound = sound.toLowerCase(Locale.ENGLISH);
-					if (!SOUND_VALID_PATTERN.matcher(sound).matches())
-						continue;
-					w.playSound(location, sound, volume, pitch);
-				} else {
-					w.playSound(location, soundEnum, volume, pitch);
-				}
+		} else if (locations != null) {
+			for (Location location : locations.getArray(event)) {
+				SoundReceiver.play(World::playSound, World::playSound, location.getWorld(),
+					location, sounds.getArray(event), category, volume, pitch);
 			}
 		}
 	}
 
 	@Override
-	public String toString(@Nullable Event e, boolean debug) {
+	public String toString(@Nullable Event event, boolean debug) {
+		StringBuilder builder = new StringBuilder()
+			.append("play sound ")
+			.append(sounds.toString(event, debug));
+		
+		if (category != null)
+			builder.append(" in ").append(category.toString(event, debug));
+		
+		if (volume != null)
+			builder.append(" with volume ").append(volume.toString(event, debug));
+		
+		if (pitch != null)
+			builder.append(" with pitch ").append(pitch.toString(event, debug));
+		
 		if (locations != null)
-			return "play sound " + sounds.toString(e, debug) +
-					(category != null ? " in " + category.toString(e, debug) : "") +
-					(volume != null ? " at volume " + volume.toString(e, debug) : "") +
-					(pitch != null ? " at pitch " + pitch.toString(e, debug) : "") +
-					(locations != null ? " at " + locations.toString(e, debug) : "") +
-					(players != null ? " for " + players.toString(e, debug) : "");
-		else
-			return "play sound " + sounds.toString(e, debug) +
-					(volume != null ? " at volume " + volume.toString(e, debug) : "") +
-					(pitch != null ? " at pitch " + pitch.toString(e, debug) : "") +
-					(players != null ? " to " + players.toString(e, debug) : "");
+			builder.append(" at ").append(locations.toString(event, debug));
+		
+		if (players != null)
+			builder.append(" to ").append(players.toString(event, debug));
+		
+		return builder.toString();
+	}
+	
+	@FunctionalInterface
+	private interface SoundReceiver<T, S> {
+		void play(@NotNull T receiver, @NotNull Location location, @NotNull S sound,
+	            @NotNull SoundCategory category, float volume, float pitch);
+		
+		static <T> void play(@NotNull SoundReceiver<T, String> stringReceiver,
+                @NotNull SoundReceiver<T, Sound> soundReceiver,
+                @NotNull T receiver, @NotNull Location location, @NotNull String[] sounds,
+                @NotNull SoundCategory category, float volume, float pitch
+		) {	
+			for (String sound : sounds) {
+				try {
+					Sound enumSound = Sound.valueOf(sound.toUpperCase(Locale.ENGLISH));
```

Comment:
No since it's up to the user to choose between Spigot and Minecraft names. We support both for compatibility, but do not automatically convert malformed names.